src/python/gudhi/weighted_rips_complex.py

```diff
@@ -0,0 +1,57 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Raphaël Tinarrage and Yuichi Ike
+#
+# Copyright (C) 2020 Inria, Copyright (C) 2020 FUjitsu Laboratories Ltd.
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from gudhi import SimplexTree
+
+class WeightedRipsComplex:
+    """
+    class to generate a weighted Rips complex 
+    from a distance matrix and filtration value
+    """
+    def __init__(self, 
+                distance_matrix=None, 
+                filtration_values=None,
+                max_filtration=float('inf')):
+        """
+        Parameters:
+            distance_matrix: list of list of float,
+                distance matrix (full square or lower triangular)
+            filtration_values: list of float,
+                flitration value for each index
+            max_filtration: float,
+                specifies the maximal filtration value to be considered        
+        """
+        self.distance_matrix = distance_matrix
+        self.filtration_values = filtration_values
+        self.max_filtration = max_filtration
+            
+    def create_simplex_tree(self, max_dimension):
+        """
+        Parameter:
+            max_dimension: int
+                graph expansion until this given dimension
+        """
+        dist = self.distance_matrix
+        F = self.filtration_values
+        num_pts = len(dist)
+        
+        st = SimplexTree()
+        
+        for i in range(num_pts):
+            if F[i] < self.max_filtration:
+                st.insert([i], F[i])
+        for i in range(num_pts):
+            for j in range(num_pts):
+                value = (dist[i][j] + F[i] + F[j]) / 2
```

Comment:
That's not going to work if your input is a triangular matrix.

Reply:
Fixed this bug. The current version gives 0-weight for each vertex if `weights` are not specified.  
Should we remove `/ 2` in value to have consistency with `RipsComplex`?

Reply:
Yes, removing /2 seems sensible.

Reply:
I thought the filtration value for `[i,j]` is not necessarily `(dist[i][j] + F[i] + F[j]) / 2` for general weights (i.e., non-DTM weights) and fixed it, adding one test. 
I think if I remove `/ 2` I also need to set the filtration value for `[i]` to be `2*F[i]`, but it is somewhat strange. I do not find the best way to modify it.